### DISCUSSION TITLE
How to send proactive messages with the help of continueConversationAsync method

### DISCUSSION THREAD

**Discussion by Jyothsna-2023** [microsoft/hack-together-teams#99] [Q&A]
Hi Team, I'm trying to send proactive messages in my bot. For this use case I have chosen the Teams toolkit - Basic bot template. Here is the code which is getting used to send the proactive message.

const currentUser = context.activity.from.id;
const conversationReferences[currentUser] = TurnContext.getConversationReference(context.activity);
const adapter = context.adapter;
await adapter.continueConversationAsync(config.botId, conversationReferences[currentUser], async (context) => {
       await context.sendActivity("Hello");
});

The bot send the message for the first time and it get crashes. Can I know the proper way to get/pass on the turncontext to the bot to send the proactive messages.

Appreciate your help!

**Comment by aycabas** [comment]
Hi @Jyothsna-2023, can you share more details about the error you are getting when the app crashes.

Here is the documentation you can find more info about sending proactive messages with bots: https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/send-proactive-messages?tabs=javascript

Sample code reference: https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/graph-proactive-installation/nodejs/bots/proactiveBot.js#L59

**Reply by Jyothsna-2023** [reply]
Thanks @aycabas for the quick response. 
I'm trying to send proactive messages on recurring basis for that I have used the toad-scheduler. Here is the code which get used to send the proactive messages.
 const task = new AsyncTask(
        'simpleTask',
        () => {
          try {
            console.log("hello ");
            adapter.continueConversationAsync(config.botId, conversationReferences[currentUser], async (context) => {
              context.sendActivity("proactive message from bot");
            });
          }
          catch (err) {
            console.log(err);
          }
          return Promise.resolve();
        }
      );
const job = new SimpleIntervalJob(
        { minutes: this.timerObj.varTimer, runImmediately: true },
        task,
        {
          id: 'id_1',
          preventOverrun: true,
        }
      );
      reminderScheduler.addSimpleIntervalJob(job);

Error screenshot:
![image](https://github.com/microsoft/hack-together-teams/assets/136434213/5937d286-a458-47ec-b0bd-252fc0e958b4)

**Reply by Jyothsna-2023** [reply]
I tried in separate method and end up with different error
![image](https://github.com/microsoft/hack-together-teams/assets/136434213/0348e6fc-b2f5-47d8-a4d5-84f9a8686ea6)
![image](https://github.com/microsoft/hack-together-teams/assets/136434213/1a05e512-6264-4a54-99df-b3016e3c8399)


By looking at both the errors, I understand that the turncontext object gets disposed after the message sent and I'm not able to figure out the way to get the valid turnContext object.

**Comment by Hisakaq8** [comment]
#70
